charts/redpanda/templates/_helpers.tpl

```diff
@@ -33,6 +33,17 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- end -}}
 {{- end -}}
 
+{{/*
+Create a default service name
+*/}}
+{{- define "redpanda.servicename"}}
+{{- if .Values.service.name}}
+{{- .Values.service.name | trunc 63 | trimSuffix "-" -}}
+{{- else }}
+{{ include "redpanda.fullname" . | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+{{- end}}
```

Comment:
```suggestion
{{- define "redpanda.servicename" -}}
{{- if dig "service" "name" false .Values -}}
{{- .Values.service.name | trunc 63 | trimSuffix "-" -}}
{{- else -}}
{{ include "redpanda.fullname" . | trunc 63 | trimSuffix "-" -}}
{{- end -}}
{{- end -}}
```
If we use dig here, we won't need to require that the service key exists.

Also, added trailing `-` to match our style for named templates.

Reply:
Thank you for your suggestion.